### PR TITLE
RFC for gateway api metrics exporter

### PR DESCRIPTION
**Pull request by david-martin**
Related to https://github.com/Kuadrant/kuadrant-operator/issues/675 and https://github.com/Kuadrant/gateway-api-state-metrics/issues/1

### CODE REVIEW

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (maleck13) [inline rfcs/0000-gateway-api-metrics-exporter.md:138]: So why are we not doing this work directly in gateway api state metrics. I am missing this piece . It seems we are starting with a new library?

Comment (maleck13) [reply rfcs/0000-gateway-api-metrics-exporter.md:138]: I see it is covered below

**Review by maleck13 — APPROVED** [review]
individual listener and httproute status metrics seems highly valuable. Shame it is considered out of reach to add to the existing implementation.

**Comment by david-martin** [comment]
@jasonmadigan @R-Lawton 
It would be great to get a review & approval on this before merging

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (jasonmadigan) [inline rfcs/0000-gateway-api-metrics-exporter.md:11]:
```suggestion
A new sub-project for a [prometheus exporter](https://prometheus.io/docs/instrumenting/writing_exporters/) that exports metrics about the state of [Gateway API resources](https://gateway-api.sigs.k8s.io/api-types/gateway/) in a Kubernetes cluster.
```

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (jasonmadigan) [inline rfcs/0000-gateway-api-metrics-exporter.md:114]: probably stupid question: assume we'll expose these listener condition types as-is, right? i.e. we won't group stuff like `Pending`, `Detached` etc. into other

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (jasonmadigan) [inline rfcs/0000-gateway-api-metrics-exporter.md:160]:
```suggestion
However, it will supersede the gateway-api-state-metrics project and dependency on kube-state-metrics.
```

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (jasonmadigan) [inline rfcs/0000-gateway-api-metrics-exporter.md:170]:
```suggestion
- potential complexity of implementing generic CEL support in that project
```

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (jasonmadigan) [inline rfcs/0000-gateway-api-metrics-exporter.md:122]:
```suggestion
gatewayapi_httproute_status_parents_conditions{namespace="<NAMESPACE>",name="<GATEWAY>",controller_name="<CONTROLLER_NAME>","parent_group="<PARENT_GROUP>",parent_kind="<PARENT_KIND>",parent_name="<PARENT_NAME>",parent_namespace="<PARENT_NAMESPACE>",type="<ResolvedRefs|Accepted|Other>} 1
```

**Review by jasonmadigan — COMMENTED** [review]
few small comments

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (david-martin) [reply rfcs/0000-gateway-api-metrics-exporter.md:114]: Yes, exactly. No smarts on grouping. Dump the type as is

**Review by jasonmadigan — APPROVED** [review]
LGTM

**Comment by R-Lawton** [comment]
👀

File: rfcs/0000-gateway-api-metrics-exporter.md
Comment (R-Lawton) [inline rfcs/0000-gateway-api-metrics-exporter.md:114]: Wdyt about using a gateway API acronym so something like gwapi for the metric name? im thinking long run having to type these out might get tedious. 
```suggestion
gwapi_gateway_status_listeners_conditions{namespace="<NAMESPACE>",name="<GATEWAY>",listener_name="<LISTENER_NAME>",type="<ResolvedRefs|Ready|Other>} 1
```

Comment (david-martin) [reply rfcs/0000-gateway-api-metrics-exporter.md:114]: I do like the shorter version.
My concern with changing the name of the metrics is that they lose [backwards compatibility](https://github.com/Kuadrant/architecture/pull/101/files#diff-c3c9a29408a0e64d785783106086dd38d7db6e8aac81f282eae480fdc1820974R109) with gateway-api-state-metrics, and any example alerts and dashboards provided as part of that. Same for the kuadrant example alerts and dashboards.
We would need to allow for migration of versions.
Do you think it's worth the change?

Comment (R-Lawton) [reply rfcs/0000-gateway-api-metrics-exporter.md:114]: yes good point, i think it would be nice to have but the efforts to change it arent worth it

**Review by R-Lawton — APPROVED** [review]

**Comment by david-martin** [comment]
Thanks @maleck13 @jasonmadigan @R-Lawton

**Pull request merged**
